### ISSUE TITLE
Implement halftoning

### ISSUE THREAD

**Issue by nclm** [tfuxu/Halftone#13]
Great app! 

With this name and this logo however, I would expect to be able to use it for doing actual halftones (the print kind), and not dithering (the screen kind). So I suggest that halftoning could be added, even maybe as a default algorithm?

There might already be open source code for it that could be integrated?

For a quick implementation, I suggest the “Color Amount” option could have two choices, either “1” (in which case it would be doing a monochrome halftoning of the grayscale image), or “4” (in which case it would do overlayed CMYK halftones).

I don’t have anything than Photoshop underhand right now, but for reference, here are the options for the default monochrome halftone it has: 

<img width="474" alt="half1" src="https://github.com/tfuxu/Halftone/assets/923008/ebf2dc2b-e320-40bb-b34c-6f75b784c8ee">

And for the CMYK halftone:

<img width="375" alt="helf2" src="https://github.com/tfuxu/Halftone/assets/923008/42c97a9b-4139-4f1d-8202-0d6e61f1a9b9">

Ideally choosing angles, shape, colours and resolution would be amazing, but a simple implementation with default angles could work too at least for a start :)

**Comment by tfuxu** [comment]
Yeah, its pretty ironic 😄️ I'll work on implementing halftone effect in the near future, as I'm planning to implement more dither algorithms pretty soon.
